File: Java/Daily098_easy/ArithmeticTables.java
/**
 * ARITHMETIC TABLES
 *
 * challenge #98 (easy)
 * http://www.reddit.com/r/dailyprogrammer/comments/zx8vw/9152012_challenge_98_easy_arithmetic_tables/
 *
 * sarthak7u@gmail.com
 */

class ArithmeticTables
{
    static void print_tables(int num, String operator)
    {
        // check if operator valid
        String[] valid_operators = {"+", "-", "*", "/"};
        for (int i = 0; i < 4; i++)
        {
            // if operator is present break
            if (valid_operators[i] == operator)
            {
                break;
            }

            // operator invalid, give error message and exit
            if (i == 3)
            {
                System.out.print("Operator is invalid!");
                return;
            }
        }

        // store nums from 0 to n
        int[] nums = new int[num + 1];
        for (int i = 0; i <= num; i++)
        {
            nums[i] = i;
        }

        // print first row
        System.out.print(" " +  operator + " ");
        System.out.print(" |  ");
        for (int i = 0; i <= num; i++)
        {
            System.out.print(nums[i]+ " ");
        }

        // print the divider
        System.out.print("\n-----------------\n");

        // print the rest of the table
        for (int i = 0; i <= num; i++)
        {
            System.out.print(" " +  i + " ");
            System.out.print(" |  ");
            for (int j = 0; j <= num; j++)
            {
                switch (operator)
                {
                    case "+":
                        System.out.print(i + j + " ");
                        break;
                    case "-":
                        System.out.print(i - j + " ");
                        break;
                    case "*":
                        System.out.print(i * j + " ");
                        break;
                    case "/":
                        System.out.print(i / j + " ");
                        break;
                }
            }
            System.out.print("\n");
        }

    }
}

File: Python/Daily098_easy/ArithmeticTables.py
## ARITHMETIC TABLES
##
## challenge #98 (easy)
## http://www.reddit.com/r/dailyprogrammer/comments/zx8vw/9152012_challenge_98_easy_arithmetic_tables/
##
##
## sarthak7u@gmail.com
##

# prints the heading row
def printFirstRow(sym, num):
    print sym, " | ",
    for i in xrange(0,num+1):
        print i,
    print ""
    line =""
    for i in xrange(0,2*(num+1+5)):
        line += "_"
    print line

# prints all rows except heading row
def printRow(numList, opNumList):
    index = 0
    for i in numList:
        print i, " | ",
        for j in opNumList[index]:
            print j,
        index += 1
        print ""
# takes required inputs from user
symbol = str(raw_input("Enter the operation: "))
num = int(raw_input("Enter a number :"))

# make list of numbers form inputted number
numbers = []
for i in xrange(0,num+1):
    numbers.append(i)

# performs operation and stores it in an array
op = []
index = 0
for i in numbers:
    op.append([])
    for j in numbers:
    # if-elif-else block to determine operation
    if symbol == "+":
        op[index].append(i+j)
    elif symbol == "*":
        op[index].append(i*j)
    elif symbol == "-":
        op[index].append(i-j)
    elif symbol == "/":
        op[index].append(i/j)
    else:
        print "operation not recognized"
        raw_input("Press enter to exit program")
        exit(1)
    index += 1

# calls print functions to print output in desired format
printFirstRow("+", num)
printRow(numbers, op)
